### ISSUE TITLE
Gzip compression indicated in Task message when using zlib compression

### ISSUE THREAD

**Issue by Gregory-Meyer** [celery/celery#4917]
## Checklist

- [x] I have included the output of ``celery -A proj report`` in the issue.
      (if you are not able to do this, then at least specify the Celery
       version affected).
- [x] I have verified that the issue exists against the `master` branch of Celery.

## Steps to reproduce

`app.py`:

```py
import celery

app = celery.Celery('tasks')

app.conf.update(
    broker_url="amqp://localhost",
    task_compression="zlib", # or "gzip"
)

if __name__ == '__main__':
    app.start()

```

`tasks.py`:

```py
from app import app


@app.task
def add(x, y):
    return x + y
```

`add.py`:

```py
#!/usr/bin/env python3
import tasks


def main():
    tasks.add.apply_async(args=(0, 0), compression="zlib") # or compression="gzip"


if __name__ == '__main__':
    main()
```

In a unix shell:

```bash
pip install -e git+git@github.com:celery/celery.git#egg=celery
rabbitmq-server &
chmod +x add.py
./add.py
```

Now inspect the contents of the `celery` queue using the RabbitMQ Management console.

## Expected behavior

If the body were specified to be compressed with gzip:

```
body: H4sIAAAAAAAA/4uONtBRMIjVUaiuBWKl5MScnKTE5OxiJSuFvNKcHB0FpdSiIjSR5IzEzDxkbn5RCpRbGwsAmiXCxFEAAAA=
headers:
  compression: application/gzip
```

If the body were specified to be compressed with zlib:

```
body: eJyLjjbQUTCI1VGorgVipeTEnJykxOTsYiUrhbzSnBwdBaXUoiI0keSMxMw8ZG5+UQqUWxsLAAjqGe0=
headers:
  compression: application/zlib
```

## Actual behavior

```
body: eJyLjjbQUTCI1VGorgVipeTEnJykxOTsYiUrhbzSnBwdBaXUoiI0keSMxMw8ZG5+UQqUWxsLAAjqGe0=
headers:
  compression: application/x-gzip
```

In summary, Celery does not use gzip at all, but the MIME type of messages indicates that it does. Celery should use gzip when requested (with the MIME type to match) and use zlib when requested (with the MIME type to match). The current behavior of indicating gzip in the MIME encoding type while actually using zlib - regardless of the choice of zlib or gzip - is very confusing to duplicate in other Celery clients.

**Comment by Gregory-Meyer** [comment]
[Related issue with Kombu](https://github.com/celery/kombu/issues/707)

**Comment by WillNye** [comment]
@Gregory-Meyer Did you find a solution?

**Comment by Gregory-Meyer** [comment]
The solution is likely wrapped up in Kombu and Worker logic. I haven't looked into it any further than the issue itself.

**Comment by ruolin-spring** [comment]
This issue does not seem to be resolved. Following directions from https://docs.celeryproject.org/en/stable/userguide/calling.html#compression, the `some_task.apply_async(args=[arg1, arg2], compression='gzip')` call is not able to have its arguments decompressed on the Worker side. Investigating the `headers` shows that the `compression` field is labeled as `application/x-gzip`. Any chance for this issue to be reopened and fixed?
